nucypher/blockchain/eth/deployers.py

```diff
@@ -293,90 +292,224 @@ class PolicyManagerDeployer(ContractDeployer):
+        if not is_checksum_address(beneficiary_address):
+            raise self.ContractDeploymentError("{} is not a valid checksum address.".format(beneficiary_address))
+        txhash = self.principal_contract.functions.transferOwnership(beneficiary_address).transact({'from': self.deployer_address})
+        self.blockchain.wait_for_receipt(txhash)
+        self.__beneficiary_address = beneficiary_address
+        return txhash
+
+    def initial_deposit(self, value: int, duration: int) -> dict:
+        """Allocate an amount of tokens with lock time, and transfer ownership to the beneficiary"""
+        # Approve
+        allocation_transactions = dict()
+        approve_txhash = self.token_agent.approve_transfer(amount=value,
+                                                           target_address=self.principal_contract.address,
+                                                           sender_address=self.deployer_address)
+        allocation_transactions['approve'] = approve_txhash
+        self.blockchain.wait_for_receipt(approve_txhash)
+
+        # Deposit
+        txhash = self.principal_contract.functions.initialDeposit(value, duration).transact({'from': self.deployer_address})
+        allocation_transactions['initial_deposit'] = txhash
+        self.blockchain.wait_for_receipt(txhash)
+        return txhash
+
+    def enroll_principal_contract(self):
+        if self.__beneficiary_address is NO_BENEFICIARY:
+            raise self.ContractDeploymentError("No beneficiary assigned to {}".format(self.principal_contract.address))
+        self.__allocation_registry.enroll(beneficiary_address=self.__beneficiary_address,
+                                          contract_address=self.principal_contract.address,
+                                          contract_abi=self.principal_contract.abi)
+
+    def deliver(self, value: int, duration: int, beneficiary_address: str) -> dict:
+        """
+        Transfer allocated tokens and hand-off the contract to the beneficiary.
+
+         Encapsulates three operations:
+            - Initial Deposit
+            - Transfer Ownership
+            - Enroll in Allocation Registry
+
+        """
+
+        deposit_txhash = self.initial_deposit(value=value, duration=duration)
+        assign_txhash = self.assign_beneficiary(beneficiary_address=beneficiary_address)
+        self.enroll_principal_contract()
+        return dict(deposit_txhash=deposit_txhash, assign_txhash=assign_txhash)
 
     def deploy(self) -> dict:
+        """Deploy a new instance of UserEscrow to the blockchain."""
+
         self.check_ready_to_deploy(fail=True, check_arming=True)
 
-        deployment_args = [self.token_agent.contract_address,
-                           self.miner_agent.contract_address,
-                           self.policy_agent.contract_address]
+        deployment_transactions = dict()
+        proxy_contract = self.__proxy_deployer.get_latest_version(blockchain=self.blockchain)
```

Comment:
difficult to understand this place, because logically we don't need to know library contract (proxy), only linker address

Reply:
:heavy_check_mark: